tests/test_project.py

```diff
@@ -841,16 +841,19 @@ def test_schema(self):
         assert len(s) == 9
         for k in "const", "const2.const3", "a", "b.b2", "c", "d", "e.e2", "f.f2":
             assert k in s
-            assert k.split(".") in s
+            with pytest.warns(FutureWarning):
+                assert k.split(".") in s
```

Comment:
This should only warn if the key contains a dot, right? Conditional context managers are somewhat awkward. Luckily, we only need to test this for split keys if the combined key contains a dot -- otherwise the code is equivalent to the test above, `assert k in s`.
```suggestion
            if "." in k:
                with pytest.warns(FutureWarning):
                    assert k.split(".") in s
```